cuda_phase-retrieval_class/src/phase_algo.hpp

```diff
@@ -82,7 +82,18 @@ class Phase_Algo
         }
     
     public:
+        Phase_Algo(pybind11::array_t<T, pybind11::array::c_style> image, pybind11::array_t<T, pybind11::array::c_style> masks, std::string mode, T beta)
+        {
+            pybind11::array_t<T, pybind11::array::c_style> randoms = pybind11::array_t<T, pybind11::array::c_style>(0);
+            init(image, masks, mode, beta, randoms);
+        }
+        
         Phase_Algo(pybind11::array_t<T, pybind11::array::c_style> image, pybind11::array_t<T, pybind11::array::c_style> masks, std::string mode, T beta, pybind11::array_t<T, pybind11::array::c_style> randoms)
+        {
+            init(image, masks, mode, beta, randoms);
+        }
+
+        void init(pybind11::array_t<T, pybind11::array::c_style> image, pybind11::array_t<T, pybind11::array::c_style> masks, std::string mode, T beta, pybind11::array_t<T, pybind11::array::c_style> randoms)
```

Comment:
Looks good. Maybe you can split up the `init()` function in smaller functions, which does only on (category of) job.

e.g.
```C++
// functions should be private
allocate_memory(); // can also parse/create some variables the like the images size
get_magnitude();
create_initial_phase();
```

Reply:
Is it okay if some of local variables become class variables?

Reply:
Yes, it is okay. We need only to pass the big gpu memory chunks as parameter, when we run `iteration()`.